Lua backports from the upstream

macbre opened the pull request:
[PLATFORM-2331](https://wikia-inc.atlassian.net/browse/PLATFORM-2331)

More backporting from Scribuntu upstream.

Fixes

```
Błąd Lua w module „ustring.lua”, w linii 325: attempt to perform arithmetic on field '?' (a nil value).
Backtrace:
(tail call): ?
ustring.lua:325: ?
(tail call): ?
mw.text.lua:225: in function "(for generator)"
mw.text.lua:212: in function "split"
Module:Disassemble:255: in function "func"
Module:Infobox:833: in function "cleanParams"
Module:Disassemble:43: in function "chunk"
mw.lua:430: in function "chunk"
(tail call): ?
[C]: in function "xpcall"
MWServer.lua:73: in function "handleCall"
MWServer.lua:266: in function "dispatch"
MWServer.lua:33: in function "execute"
mw_main.lua:7: in main chunk
[C]: ?
```

@Grunny

macbre committed:
mw.ustring functions should accept numbers where string functions do

Lua's string functions tend to auto-convert numbers to strings. We
should do the same in mw.ustring.

Backporting https://github.com/wikimedia/mediawiki-extensions-Scribunto/commit/bf39827980199f08f09de88f2a941bf6b06c160d

macbre committed:
Fix ustring errors

* mw.ustring.sub( '', 1 ) errors in LuaStandalone
* Default value for ustring.maxStringLength and ustring.maxPatternLength
  should be infinity, not nil
* mw.ustring.find() returns one value instead of two in "plain" mode.

https://github.com/wikimedia/mediawiki-extensions-Scribunto/commit/fcef54e9d98368d94816e5719b8b90cbd257351d

Grunny commented on extensions/Scribunto/engines/LuaCommon/lualib/ustring/ustring.lua:
```diff
@@ -33,6 +33,9 @@ local function checkType( name, argidx, arg, expecttype, nilok )
 end
 
 local function checkString( name, s )
+    if type( s ) == 'number' then
```
Minor: all these have inconsistent whitespace on the first line of the added blocks.

Grunny commented:
LGTM.

macbre committed:
Fix mw.ustring edge cases

A few edge cases were being incorrectly handled:
* mw.ustring.sub( 'abc', 1, 0 ) returned 'a', not ''.
* mw.ustring.codepoint( 'abc', 1, 0 ) returned 97, not no results.
* mw.ustring.codepoint( 'abc', 4, 4 ) returned 99, not no results.
* mw.ustring.gcodepoint had the same issues as mw.ustring.codepoint.

https://github.com/wikimedia/mediawiki-extensions-Scribunto/commit/d8314539daa86b000de0ad603b967e91563a07f8

Grunny commented:
:+1:

macbre committed:
ustring: fix whitespaces

macbre commented:
Backported to release in aa350d8
